### PR TITLE
JAMES-3729 LocalDelivery error handling is non-standard

### PR DESCRIPTION
**Pull request by vttranlina**
jira: https://issues.apache.org/jira/browse/JAMES-3729

### CODE REVIEW

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/LocalDelivery.java [hunk 1]
```diff
@@ -77,6 +77,8 @@ public void init() throws MessagingException {
                 .metric(metricFactory.generate(LOCAL_DELIVERED_MAILS_METRIC_NAME))
                 .build())
             .consume(getInitParameter("consume", true))
+            .ignoreError(getInitParameter("ignoreError", false))
+            .errorProcessor(getInitParameter("errorProcessor", Mail.ERROR))
```
Comment (chibenwa) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/LocalDelivery.java:81]: Please use the standard `onMailetException` property to do this.

CF https://james.staged.apache.org/james-distributed-app/3.7.0/configure/mailetcontainer.html#_error_handling

**Comment by vttranlina** [comment]
I'm not sure when `onMailetException=propagate`

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/delivery/MailDispatcher.java [hunk 2]
```diff
@@ -109,12 +101,12 @@ public MailDispatcher build() {
...
 
-    private MailDispatcher(MailStore mailStore, MailetContext mailetContext, boolean consume, boolean ignoreError, String errorProcessor) {
+    private MailDispatcher(MailStore mailStore, MailetContext mailetContext, boolean consume, String errorProcessor) {
         this.mailStore = mailStore;
         this.consume = consume;
         this.mailetContext = mailetContext;
         this.errorProcessor = errorProcessor;
-        this.ignoreError = ignoreError;
+        this.ignoreError = errorProcessor.equalsIgnoreCase("ignore");
     }
```
Comment (quantranhong1999) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/delivery/MailDispatcher.java:110]: I think we can remove the ignore/propagate parsing. Our mailet pipeline already handles that. cf `ProcessorImpl`

Comment (quantranhong1999) [reply server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/delivery/MailDispatcher.java:110]: not sure if we need to customize the onMailetException handling though

Comment (vttranlina) [reply server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/delivery/MailDispatcher.java:110]: yes, it looks like a customized

**Review by Arsnael — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]
Please add a test for the default behaviour when there is an error, if it is not tested yet ;-)

**Comment by vttranlina** [comment]
> Please add a test for the default behaviour when there is an error, if it is not tested yet ;-)

It's already `org.apache.james.transport.mailets.delivery.MailDispatcherTest#errorsShouldBeWellHandled`

**Pull request merged**